src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,79 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
 public class SalaryInfo {
+    private Map<String, Integer> nameSalaryMap = new LinkedHashMap<>();
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
+        if (names == null || data == null || dateFrom == null || dateTo == null) {
+            return null;
+        }
+
+        Map<String, Integer> resultInfoMap = new LinkedHashMap<>();
+
+        LocalDate dateStart = findDate(dateFrom);
+        LocalDate dateEnd = findDate(dateTo);
+        StringBuffer stringBufferFull = new StringBuffer();
+        StringBuffer stringBufferEmpty = new StringBuffer();
+        stringBufferFull.append("Report for period " + dateFrom + " - " + dateTo);
+        stringBufferEmpty.append("Report for period " + dateFrom + " - " + dateTo);
+
+        for (int i = 0; i < names.length; i++) {
+            nameSalaryMap.put(names[i], 0);
+            int resultSalary = 0;
+            for (int j = 0; j < data.length; j++) {
+                if ((findDate(data[j]).compareTo(dateStart) > 0
+                        && findDate(data[j]).compareTo(dateEnd) < 0)
+                        || findDate(data[j]).equals(dateStart)
+                        || findDate(data[j]).equals(dateEnd)) {
+                    if (data[j].contains(names[i])) {
+                        String dataEmployee = data[j].substring(data[j].indexOf(names[i]));
+                        String[] hourSalary = dataEmployee.split(" ");
+                        int hour = Integer.parseInt(hourSalary[1]);
+                        int salaryValue = Integer.parseInt(hourSalary[2]);
```

Comment:
for all indexes
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/d1ea19a9-68ab-426f-a458-8e771f8c3464)

Reply:
not fixed - add constants for indexes

Reply:
Only calculate salary inside IF - no any StringBuilder or String